### PR TITLE
Issue 7370: Add an integration test for SegmentReader.

### LINKED ISSUE #7370: Add an integration test for SegmentReader.
**Is your feature request related to a problem? Please describe.**
Add an integration test for SegmentReader.

**Describe the solution you'd like**
Add an integration test to cover the newly added segment reader API.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Bhupender-Y**
**Change log description**  
Add an integration test for SegmentReader.

**Purpose of the change**  
 Closes #7370 

**What the code does**  
This integration test will verify the segment reader functionality.

**How to verify it**  
Integration test should be successful.

### CODE REVIEW

File: test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+        SegmentReader<Integer> segmentReader = segmentReaderList.get(0);
+        log.info("Starting reading the events.");
+        boolean exitNow = false;
+        while (!exitNow) {
+            try {
+                segmentReader.read(timeout);
+                readEventCount++;
+            } catch (EndOfSegmentException e) {
+                exitNow = true;
```
Comment (tkaitchuck) [inline test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java:164]: break?

File: test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        SegmentReader<Integer> segmentReader = segmentReaderList.get(0);
+
+        boolean exitNow = false;
+        while (!exitNow) {
+            try {
+                segmentReader.read(timeout);
+                readEventCount++;
+            } catch (EndOfSegmentException e) {
+                log.warn("End of segment reached");
+                exitNow = true;
```
Comment (tkaitchuck) [inline test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java:207]: break?

File: test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java (hunk 1 above)
Comment (Bhupender-Y) [reply test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java]: Done

File: test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java (hunk 2 above)
Comment (Bhupender-Y) [reply test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java]: Done

File: test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+        streamCut = new StreamCutImpl(Stream.of(scope, stream), ImmutableMap.of(new Segment(scope, stream, 0L), 0L));
+
+        log.info("Creating segment reader manager.");
+        @Cleanup
+        SegmentReaderManager<Integer> segmentReaderManager = SegmentReaderManager.create(clientConfig, serializer);
+        //Try to read the segment from offset 0.
+        List<SegmentReader<Integer>> segmentReaderList = segmentReaderManager.getSegmentReaders(Stream.of(scope, stream), streamCut).get();
+        assertEquals(1, segmentReaderList.size());
+        SegmentReader<Integer> segmentReader = segmentReaderList.get(0);
```
Comment (AnkurAnand11) [inline test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java:197]: Why Integer ?

Comment (Bhupender-Y) [reply test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java]: Thanks, i have updated the Serializer and also add assert to verify the events.

**Comment by AnkurAnand11** [comment]
Can we have a test case for multiple segments ?

**Comment by Bhupender-Y** [comment]
> Can we have a test case for multiple segments ?

@AnkurAnand11 Yes, i have added a test for stream with multiple segments.

**Review by tkaitchuck — APPROVED** [review]

**Pull request merged**